--- src/buildroot-2017.11.1/package/Config.in	2021-06-23 10:28:35.748051977 -0500
+++ buildroot-2017.11.1/package/Config.in	2021-06-23 10:29:08.964249418 -0500
@@ -1733,6 +1733,7 @@
 	source "package/open-plc-utils/Config.in"
 	source "package/openntpd/Config.in"
 	source "package/openobex/Config.in"
+	source "package/openresolv/Config.in"
 	source "package/openssh/Config.in"
 	source "package/openswan/Config.in"
 	source "package/openvpn/Config.in"
