--- src/buildroot-2017.11.1/package/Config.in	2017-12-31 03:03:52.000000000 -0600
+++ buildroot-2017.11.1/package/Config.in	2021-04-05 21:56:26.770366644 -0500
@@ -1806,7 +1806,8 @@
 	source "package/wavemon/Config.in"
 	source "package/wget/Config.in"
 	source "package/whois/Config.in"
-	source "package/wireguard/Config.in"
+        source "package/wireguard-linux-compat/Config.in"
+        source "package/wireguard-tools/Config.in"
 	source "package/wireless-regdb/Config.in"
 	source "package/wireless_tools/Config.in"
 	source "package/wireshark/Config.in"
